src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -130,9 +130,31 @@ define(function (require, exports, module) {
     }
 
     // Get Analytics data
-    function getAnalyticsData() {
+    function getAnalyticsData(EventParams) {
         var userUuid = PreferencesManager.getViewState("UUID"),
             olderUuid = PreferencesManager.getViewState("OlderUUID");
+        var eventParams = {};
+
+        if (!EventParams || EventParams === undefined) {
+            eventParams.eventCategory = "pingData";
+            eventParams.eventSubCategory = "";
+            eventParams.eventType = "";
+            eventParams.eventSubType = "";
+            EventParams = eventParams;
+        } else {
+            if (!EventParams.eventCategory || EventParams.eventCategory === undefined) {
```

Comment:
!EventParams.eventCategory will take care of undefined. 
same in following blocks